Embedded/MaxMix/Display.ino

```diff
@@ -333,5 +430,51 @@ void DrawSelectionVolumeBar(Adafruit_SSD1306* display, uint8_t volume, bool isMu
   x0 = DISPLAY_AREA_CENTER_MARGIN_SIDE + DISPLAY_AREA_CENTER_WIDTH - 1;
 
   display->drawLine(x0, y0, x0, y1, WHITE);
+
+}
+
+//---------------------------------------------------------
+// Calculates Offset to print scrolling text
+//---------------------------------------------------------
+uint16_t ScrollOffset(uint32_t* displayScrollTimer, uint32_t now, size_t nameLength, size_t nameLengthOther, uint8_t stepDelay, uint16_t maxOffset)
+{
+  uint32_t scrollTimerDelta = now - *displayScrollTimer;
+
+  if (scrollTimerDelta < DISPLAY_SCROLL_DELAY_INITIAL)
+  {
+    return 0;
+  }
+
+  uint16_t currentOffset = (scrollTimerDelta - DISPLAY_SCROLL_DELAY_INITIAL) / stepDelay;
   
-}
\ No newline at end of file
+  if (currentOffset < maxOffset) //  Scroll
+  {
+    return currentOffset;
+  }
+  else // Reset Scroll
+  {
+    if (nameLengthOther == 0) // GameMode check
+    {
+      UpdateScrollTimer();
+    }
+    else if (nameLength >= nameLengthOther)
+    {
+      UpdateScrollTimer();
+    }
+    return 0;
+  }
+}
+
+//---------------------------------------------------------
+//---------------------------------------------------------
+uint16_t GetMaxScrollOffset(Adafruit_SSD1306* display, char* name)
+{
+  char displayName[strlen(name) + strlen(DISPLAY_SCROLL_ITEM_SEPARATOR) + strlen(name) + 1] = "";
+  strcpy(displayName, name);
+  strcat_P(displayName, DISPLAY_SCROLL_ITEM_SEPARATOR);
+  
+  uint16_t maxOffset;
+  display->getTextBounds(displayName, 0, 0, NULL, NULL, &maxOffset, NULL);
```

Comment:
So the spacing between characters is not constant, it depends on what the characters are, the font size, and font style etc. Calculating manually will cause some weird edge cases to return wrong offset/calculation values. For example, applications with multiple symbols or special characters like <tab> and <space>.

To keep things compatible with any app name even weird ones, which I think is a fair assumption that there can and will be weird names, I used **display->getTextBounds** which is part of the GFX lib.